data-reporting-service/observation-service/src/main/java/gov/cdc/etldatapipeline/observation/util/ProcessObservationDataUtil.java

```diff
@@ -0,0 +1,158 @@
+                        }
+                    }
+                    else {
+                        logger.error("obsDomainCdSt1: {} is not valid", obsDomainCdSt1);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Error processing Person Participation JSON array from observation data: {}", e.getMessage());
+        }
+    }
+
+    private void transformOrganizationParticipations(String organizationParticipations, String obsDomainCdSt1, ObservationTransformed observationTransformed, ObjectMapper objectMapper) {
+        try {
+            JsonNode organizationParticipationsJsonArray = organizationParticipations != null ? objectMapper.readTree(organizationParticipations) : null;
+
+            if(organizationParticipations != null && organizationParticipationsJsonArray.isArray()) {
+                for(JsonNode jsonNode : organizationParticipationsJsonArray) {
+                    String typeCd = jsonNode.get("type_cd").asText();
+                    String subjectClassCd = jsonNode.get("subject_class_cd").asText();
+
+                    if(obsDomainCdSt1.equals("Result")) {
+                        if(typeCd != null && subjectClassCd != null) {
+                            if(typeCd.equals("PRF") && subjectClassCd.equals("ORG")) {
+                                observationTransformed.setPerformingOrganizationId(jsonNode.get("entity_id").asLong());
+                            }
+                        }
+                        else {
+                            logger.error("typeCd or subjectClassCd is null for the organizationParticipations: {}", organizationParticipations);
+                        }
+                    }
+                    if(obsDomainCdSt1.equals("Order")) {
+                        if(typeCd != null && subjectClassCd != null) {
+                            if(typeCd.equals("AUT") && subjectClassCd.equals("ORG")) {
+                                observationTransformed.setAuthorOrganizationId(jsonNode.get("entity_id").asLong());
+                            }
+                            if(typeCd.equals("ORD") && subjectClassCd.equals("ORG")) {
+                                observationTransformed.setOrderingOrganizationId(jsonNode.get("entity_id").asLong());
+                            }
+                        }
+                        else {
+                            logger.error("typeCd or subjectClassCd is null for the organizationParticipations: {}", organizationParticipations);
+                        }
+                    }
+                    else {
+                        logger.error("obsDomainCdSt1: {} is not valid", obsDomainCdSt1);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Error processing Organization Participation JSON array from observation data: {}", e.getMessage());
```

Comment:
@rshanmugam-enquizit is it possible to get the corresponding Id / topic ?

Reply:
Yes, it is possible! This one will be fixed with the other as a part of optimization